src/benchmarks/micro/MicroBenchmarks.csproj

```diff
@@ -16,29 +16,50 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <EnableXlfLocalization>false</EnableXlfLocalization>
   </PropertyGroup>
-
+  <!-- out-of-band packages that are not included in NetCoreApp have TFM-specific versions -->
+  <Choose>
+    <When Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
+        <PropertyGroup>
+            <ExtensionsVersion>3.1.22</ExtensionsVersion>
+            <SystemVersion>4.7.1</SystemVersion>
+        </PropertyGroup>
+    </When>
+    <When Condition="'$(TargetFramework)' == 'net5.0'">
+        <PropertyGroup>
+            <ExtensionsVersion>5.0.0</ExtensionsVersion>
+            <SystemVersion>5.0.0</SystemVersion>
+        </PropertyGroup>
+    </When>
+    <Otherwise>
+        <!-- when comparing against Full .NET Framework we are usually interested in CLR differences, so net461 belongs to this block -->
```

Comment:
Would it be better to explicitly condition this block on TargetFramework=net461 (or whatever)?

If I understand right, these will be ignored for .net6.0 or later because they're in box at that point -- so that would be more explicit.
Or if they won't be ignored, then it would be also good to add the condition so things break when we test a future TFM that needs us to update this version number here?

Reply:
> If I understand right, these will be ignored for .net6.0 or later because they're in box at that point

Do you mean that `Microsoft.Extensions.Caching.Memory` and others are now in-box? That is new to me. And it makes my life easier as it means that we won't ever need to update these versions?